bip-0340.mediawiki

```diff
@@ -128,6 +128,7 @@ The following conventions are used, with constants as defined for [https://www.s
 *** Fail if ''c &ne; y<sup>2</sup> mod p''.
 *** Return the unique point ''P'' such that ''x(P) = x'' and ''y(P) = y'', or fail if no such point exists.
 ** The function ''lift_x_even_y(x)'', where ''x'' is an integer in range ''0..p-1'', returns the point ''P'' for which ''x(P) = x'' and ''has_even_y(P)'', or fails if no such point exists. If such a point does exist, it is always equal to either ''lift_x_square_y(x)'' or ''-lift_x_square_y(x)'', which suggests implementing it in terms of ''lift_x_square_y'', and optionally negating the result.
+** The function ''extbytes(P)'', where ''P'' is a point, returns the compressed 33-byte encoding of P, that is ''0x02 || bytes(x(P))'' if ''has_even_y(P)'' or ''0x03 || bytes(x(P))'' otherwise.
```

Comment:
How about `compressed(P)` instead of `extbytes(P)`. It is what it is. And extbytes can be easily confused with bip32 extended keys.

Reply:
`compressed` is strange in another way: it's larger than the ordinary encoding (which is 32-byte in the context of this BIP). `bytes_with_y` maybe?

Reply:
Having 64 byte `sk` means there's no need for this, I think, so maybe we can cleverly avoid this bikeshed? :)

Reply:
See my other comment; I think we can avoid extbytes/compressed/bytes_with_y altogether, even when not including the pubkey inside the secret key.

Reply:
Agree, I've stuck that in as an intermediate commit; the 64-byte sk still seems appealing on its own merits to me though -- and it seems easy enough for implementors to implement `Sign(KeyGen(r)[0], m)` if they want to have 32 byte secrets?